drivers/clock_control/clock_control_nrf.c

```diff
@@ -461,6 +461,15 @@ static void lfclk_spinwait(enum nrf_lfclk_start_mode mode)
 		nrf_clock_int_disable(NRF_CLOCK, NRF_CLOCK_INT_LF_STARTED_MASK);
 	}
 
+	if ((mode == CLOCK_CONTROL_NRF_LF_START_AVAILABLE) &&
+	    (target_type == NRF_CLOCK_LFCLK_Xtal) &&
+	    (nrf_clock_lf_srccopy_get(NRF_CLOCK) == CLOCK_CONTROL_NRF_K32SRC)) {
+		/* RC might already be started. If current LFCLK source equals
```

Comment:
I think the code is fine, but the comment in isolation of the commit message is misleading, because the code is not checking the current LFCLK source, but rather the configured LFCLK source, and the RC is known to be running because configuring the crystal as the source is only done after RC has started.

The minimal change would be "~current~ *configured* LFCLK source equals...", but referencing the two-stage process would also help.